File: xxf_arch/src/main/java/com/xxf/arch/permission/CameraPermissionTransformer.java
package com.xxf.arch.permission;

import android.content.Context;
import android.support.annotation.NonNull;

/**
 * @author youxuan  E-mail:youxuan@icourt.cc
 * @version 2.3.1
 * @Description 相机权限
 * @date createTime：2018/9/3
 */
public class CameraPermissionTransformer extends XXFPermissionTransformer {

    private CameraPermissionTransformer(@NonNull Context context, @NonNull String rejectNotice) {
        super(context, rejectNotice);
    }

    public CameraPermissionTransformer(@NonNull Context context, boolean rejecctJumpPermissionSetting) {
        super(context, "请开启相机权限", rejecctJumpPermissionSetting);
    }

    public CameraPermissionTransformer(@NonNull Context context) {
        super(context, "请开启相机权限");
    }
}

File: core/lib_permission/src/main/java/com/xxf/permission/transformer/CameraPermissionTransformer.kt
package com.xxf.permission.transformer

import android.Manifest
import android.content.Context

/**
 * @version 2.3.1
 * @Author: XGod  xuanyouwu@163.com  17611639080  https://github.com/NBXXF     https://blog.csdn.net/axuanqq
 * @Description 相机权限
 * @date createTime：2018/9/3
 */
@Deprecated("过时了,请用map PermissionCheckMapFunction 或者 PermissionCheckForResultMapFunction")
class CameraPermissionTransformer(context: Context) :
    RxPermissionTransformer(context, Manifest.permission.CAMERA)